src/components/molecules/Breadcrumbs/BreadcrumbsTitle.tsx

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react'
+import { useFetchUser } from 'hooks/requests/useUsers'
+import { useVendorFetch } from 'hooks/requests/useVendors'
+import { BreadcrumbComponentProps } from 'use-react-router-breadcrumbs'
+import { useTranslation } from 'react-i18next'
+
+interface idTypes {
+  vendorId?: string
+  userId?: string
+  orderId?: string
+}
+
+const BreadcrumbsTitle = <ParamKey extends string = string>({
+  match,
+}: BreadcrumbComponentProps<ParamKey>) => {
+  const { t } = useTranslation()
+
+  const { vendorId, userId, orderId }: idTypes = match?.params
+
+  const { vendor } = useVendorFetch({ id: vendorId })
+  const { user } = useFetchUser({ id: userId })
+
+  const { name } = useMemo(() => {
+    switch (true) {
```

Comment:
It is a little bit unconventional. Is there a reason not to go with  a classic if else in this case?